### PR TITLE
[Feature Anywhere] Fix legacy tests / bug fixes

### PR DESCRIPTION
**Pull request by ohltyler**
### Description

This PR gets all legacy functional tests to pass by changing the following:
- adding NPE check in the vega parser for undefined `visibleVisLayers`. Was a miss the first time
- Removing unnecessary `savedObjectId` params in the vega expression function pipeline (and downstream vega visualization) which was leftover from the old way of propagating the data down. This was causing an issue where when creating a regular vega visualization, under certain circumstances, would cause the rendering to fail due to missed parameters when generating the `vega` expression function
- Removing unnecessary calls to fetch `VisLayer`s when rendering the embeddable outside of the context of a dashboard. This is a general improvement, but also fixes a bug where it can put the embeddable in a weird state such that the inspect button is greyed out when first loading a vis in edit view (resolves after re-loading). But this fix prevents that from happening at all

### CODE REVIEW

**Comment by ohltyler** [comment]
Tests are all consistently passing locally:
```
37 passing (12.0m)
8 pending
```
Errors seen in CI:

Missing elements/visualizations:
```
     │1)    dashboard app
     │       using current data
     │         create and add embeddables
     │           add new visualization link
     │             adds new visualiztion via the top nav link:
     │
     │      retry.try timeout: Error: 3 elements completed rendering, still waiting on a total of 4
     │                 specifically:
     │ visualization from top nav add new panel

     │2)    dashboard app
     │       using current data
     │         create and add embeddables
     │           add new visualization link
     │             adds a new visualization:
     │
     │      retry.try timeout: Error: 3 elements completed rendering, still waiting on a total of 5
     │                 specifically:
     │ visualization from top nav add new panel
     │ visualization from add new link
```

Other errors seen in run:
```
                 │ERROR browser[SEVERE] http://localhost:6610/9007199254740991/bundles/plugin/visualizations/visualizations.plugin.js 6276:14 Error: Could not locate that index-pattern (id: 1b1789d0-9e93-11ea-853e-adc0effaf76d), [click here to re-create it](management/opensearch-dashboards/indexPatterns)
```
This one looks related to `augment-vis` changes, still researching:
```
               │ERROR browser[SEVERE] http://localhost:6610/internal/search/opensearch - Failed to load resource: the server responded with a status of 400 (Bad Request)
               │ERROR browser[SEVERE] http://localhost:6610/api/saved_objects/_find?default_search_operator=AND&page=1&per_page=100&search_fields=title%5E3&search_fields=description&type=augment-vis - Failed to load resource: the server responded with a status of 503 (Service Unavailable)
```

```
               │ERROR browser[SEVERE] http://localhost:6610/9007199254740991/bundles/core/core.entry.js 20506:18 TypeError: Failed to fetch
```
```
                 │ERROR browser[SEVERE] http://localhost:6610/internal/search/opensearch - Failed to load resource: the server responded with a status of 500 (Internal Server Error)
                 └- ✓ pass  (5.9s) "dashboard app using current data create and add embeddables add new visualization link saves the saved visualization url to the app link"
               └-> "after all" hook for "saves the saved visualization url to the app link"
                 │ proc [opensearch-dashboards]   log   [23:30:59.211] [error][data][opensearch] [search_phase_execution_exception]: 
                 │ proc [opensearch-dashboards]  error  [23:30:59.131]  Error: Internal Server Error
                 │ proc [opensearch-dashboards]     at HapiResponseAdapter.toError (/home/runner/work/OpenSearch-Dashboards/OpenSearch-Dashboards/src/core/server/http/router/response_adapter.ts:152:19)
```

I will try a new local stack with vanilla OSD/ES without any external plugins, such as AD, and see if there is a difference

**Comment by ohltyler** [comment]
Confirmed this still passes locally with vanilla OSD

**Comment by ohltyler** [comment]
Actually - current failures exist in `main` as well currently. For example, latest commit right now is #4268 - see [same test run failure](https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/5273048338/jobs/9538939761)

**Comment by ohltyler** [comment]
Remaining failures after first fix:
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4310#issuecomment-1599955849

**Comment by ohltyler** [comment]
Group 7 passing locally
```
48 passing (9.0m)
5 pending
```

**Comment by ohltyler** [comment]
Last completed test run: https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/5336998643

The failures seen there are equivalent to existing failures as seen on `main`. More details in related issue #4310

File: yarn.lock
Comment (ohltyler) [inline yarn.lock:1]: These changes are just what was pulled into the lockfile after re-running `bootstrap`. Prefer to include the change so it is consistent

**Review by ananzh — APPROVED** [review]
LGTM

**Review by AMoo-Miki — APPROVED** [review]

**Review by joshuarrrr — APPROVED** [review]
Nice - Thanks @ohltyler for the PR summary. It sounds like we were able to catch a few important issues.

**Pull request merged**